tests/ActivityApiTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+use Rogue\Models\Signup;
+use Rogue\Models\Event;
+use Faker\Generator;
+
+class ActivityApiTest extends TestCase
+{
+    /*
+     * Base URL for the Api.
+     */
+    protected $activityApiUrl = 'api/v2/activity';
+
+    /**
+     * Test for retrieving a user's activity with limit query param.
+     *
+     * GET /activity?limit=8
+     * @return void
+     */
+    public function testActivityIndexWithLimitQuery()
+    {
+        $response = $this->json('GET', $this->activityApiUrl . '?limit=8');
+
+        $this->assertResponseStatus(200);
+
+        $response = $this->decodeResponseJson();
+
+        $this->assertEquals(8, $response['meta']['pagination']['per_page']);
```

Comment:
This is totally subjective, but I really like the `seeJsonSubset` test method for stuff like this. It feels like it makes it a bit easier to see exactly what you're expecting the response to look like:

```php
$this->json('GET', $this->activityApiUrl . '?limit=8');

$this->assertResponseStatus(200);
$this->seeJsonSubset([
  'meta' => [
    'pagination' => [
      'per_page' => 8
    ]
  ]
]);  
```